src/ansible_compat/runtime.py

```diff
@@ -68,22 +96,301 @@ def version(self) -> Version:
             msg = "Unable to find a working copy of ansible executable."
             raise MissingAnsibleError(msg, proc=proc)
 
-        self._version = Version(version)
+        self._version = packaging.version.Version(version)
         return self._version
 
-    # def install_collection(self, collection: str) -> None:
-    #     """..."""
-    #     ...
-
-    # def install_requirements(self, requirement: str) -> None:
-    #     """..."""
-    #     ...
-
-    # def prepare_environment(
-    #     self,
-    #     project_dir: Optional[str] = None,
-    #     offline: bool = False,
-    #     required_collections: Optional[Dict[str, str]] = None,
-    # ) -> None:
-    #     """..."""
-    #     ...
+    def install_collection(
+        self, collection: str, destination: Optional[Union[str, pathlib.Path]] = None
+    ) -> None:
+        """Install an Ansible collection.
+
+        Can accept version constraints like 'foo.bar:>=1.2.3'
+        """
+        cmd = [
+            "ansible-galaxy",
+            "collection",
+            "install",
+            "-v",
+        ]
+        if destination:
+            cmd.extend(["-p", str(destination)])
+        cmd.append(f"{collection}")
+
+        _logger.info("Running %s", " ".join(cmd))
+        run = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            check=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        if run.returncode != 0:
```

Comment:
I'd not use this C-style obligatory rc check idiom. It's unpythonic and error-prone because it adds extra cognitive load on the users who have to always remember to do this check.
It's better to have `check=True` because this way (1) it forces you to do try/except and you can't just forget to handle an error + (2) you could explicitly chain the exception cause with `raise X from Y` which greatly improves the DX, specifically the respectability of tracebacks.